packages/flutter/lib/src/painting/borders.dart

```diff
@@ -374,13 +410,23 @@ class Border {
     return true;
   }
 
+  Border add(Border typedOther) {
+    if (BorderSide.canMerge(top, typedOther.top) &&
+        BorderSide.canMerge(right, typedOther.right) &&
+        BorderSide.canMerge(bottom, typedOther.bottom) &&
+        BorderSide.canMerge(left, typedOther.left)) {
+      return Border.merge(this, typedOther);
+    }
+    return null;
```

Comment:
Why not assert versus returning null?

Reply:
In the full branch, if add() returns null then a CompoundBorder is generated instead:
https://github.com/Hixie/flutter/blob/borders/packages/flutter/lib/src/painting/borders.dart#L300